src/module.c

```diff
@@ -11339,60 +11362,39 @@ long long getModuleNumericConfig(ModuleConfig *module_config) {
 
 /* This function takes a module and a list of configs stored as sds NAME VALUE pairs.
  * It attempts to call set on each of these configs. */
-int loadModuleConfigs(RedisModule *module, int is_loadex) {
+int loadModuleConfigs(RedisModule *module) {
     listIter li;
-    listIter li_supplied;
     listNode *ln;
-    listNode *ln_supplied;
-    int count = 0;
-    sds *config_pair;
     const char *err = NULL;
-
     listRewind(module->module_configs, &li);
     while ((ln = listNext(&li))) {
         ModuleConfig *module_config = listNodeValue(ln);
-        int config_supplied = 0;
         sds config_name = sdscatfmt(sdsempty(), "%s.%s", module->name, module_config->name);
-        listRewind(server.module_configs_queue, &li_supplied);
-        while ((ln_supplied = listNext(&li_supplied))) {
-            sds supplied_config = ln_supplied->value;
-            config_pair = sdssplitargs(supplied_config, &count);
-            if (!(strcasecmp(config_name, config_pair[0]))) {
-                if (!performModuleConfigSetFromName(config_pair[0], config_pair[1], &err)) {
-                    serverLog(LL_WARNING, "Issue during loading of configuration %s : %s", config_pair[0], err);
-                    sdsfreesplitres(config_pair, count);
-                    sdsfree(config_name);
-                    listEmpty(server.module_configs_queue);
-                    return REDISMODULE_ERR;
-                }
-                config_supplied = 1;
-                sdsfreesplitres(config_pair, count);
-                listDelNode(server.module_configs_queue, ln_supplied);
-                break;
+        dictEntry *config_argument = dictFind(server.module_configs_queue, config_name);
+        module_config->is_initialized = 1;
+        if (config_argument) {
+            if (!performModuleConfigSetFromName(dictGetKey(config_argument), dictGetVal(config_argument), &err)) {
+                serverLog(LL_WARNING, "Issue during loading of configuration %s : %s", (sds) dictGetKey(config_argument), err);
+                sdsfree(config_name);
+                dictEmpty(server.module_configs_queue, NULL);
+                return REDISMODULE_ERR;
             }
-            sdsfreesplitres(config_pair, count);
-        }
-        
-        if (!config_supplied) {
+        } else {
             if (!performModuleConfigInitFromName(config_name, &err)) {
```

Comment:
maybe the `performModuleConfigInitFromName` function should have the word "Default" or even "DefaultValue" in it?